sql/datetimetype.go

```diff
@@ -205,7 +205,7 @@ func (t datetimeType) ConvertWithoutRangeCheck(v interface{}) (time.Time, error)
 			return zeroTime, ErrConvertingToTime.New(v)
 		}
 	case time.Time:
-		res = value.UTC()
+		res = value
```

Comment:
@zachmu Pretty confident that we need this change. For datetime objects (time.Time) it's essential we return the object with the correct timezone loaded in. Converting to UTC consistently does not seem correct

Reply:
I actually don't think this will change anything for any tests that hit Dolt. All throughout Dolt we call `time.UTC()` to _make sure_ that we're standardizing them. There's probably other places in gms where we do this `time.UTC()` call so I'm not sure if this is even changing anything. From a conceptual level though, this particular change affects both `TIMESTAMP` and `DATETIME`. `TIMESTAMP` is definitely supposed to convert to UTC, and `DATETIME` does not.

For the purposes of `CONVERT_TZ`, this change shouldn't be necessary. When you call `time.UTC()`, Go is simply setting the internal location to UTC. It's easy to simply set it back in your `CONVERT_TZ` function (which you're going to have to do anyway). It doesn't change the internal unix timestamp at all. In fact, all locations/timezones are just for display purposes as far as Go is concerned, so converting it to UTC, finding the timezone delta, and applying that to the time gives the exact same result as far as what is exposed from MySQL.

We shouldn't be comparing times using `==`, but if we are, having everything as UTC at least ensures that all of those will evaluate correctly. We could treat UTC as "no timezone", and any "timezone modifications" would just offset the actual timestamp by some amount. There are probably more changes that would need to be made to accommodate this, but it's far easier if we can ensure all times are set to UTC to begin with.

TL;DR: this change isn't necessary and we can leave it as UTC() for the sake of consistency, as multiple places reference this. If this change is required to make your code work, then I'm not sure that code is correct.